src/main/java/net/minecraftforge/fmp/multipart/MultipartContainer.java

```diff
@@ -0,0 +1,751 @@
+package net.minecraftforge.fmp.multipart;
+
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.Collection;
+import java.util.HashMap;
+import java.util.Iterator;
+import java.util.LinkedHashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Map.Entry;
+import java.util.Random;
+import java.util.UUID;
+
+import com.google.common.collect.BiMap;
+import com.google.common.collect.ImmutableBiMap;
+
+import io.netty.buffer.ByteBuf;
+import io.netty.buffer.Unpooled;
+import net.minecraft.block.Block;
+import net.minecraft.block.material.Material;
+import net.minecraft.entity.Entity;
+import net.minecraft.entity.player.EntityPlayer;
+import net.minecraft.item.ItemStack;
+import net.minecraft.nbt.NBTTagCompound;
+import net.minecraft.nbt.NBTTagList;
+import net.minecraft.network.PacketBuffer;
+import net.minecraft.tileentity.TileEntity;
+import net.minecraft.util.EnumFacing;
+import net.minecraft.util.EnumHand;
+import net.minecraft.util.ResourceLocation;
+import net.minecraft.util.math.AxisAlignedBB;
+import net.minecraft.util.math.BlockPos;
+import net.minecraft.util.math.RayTraceResult;
+import net.minecraft.util.math.Vec3d;
+import net.minecraft.world.IBlockAccess;
+import net.minecraft.world.World;
+import net.minecraftforge.common.MinecraftForge;
+import net.minecraftforge.common.capabilities.Capability;
+import net.minecraftforge.common.capabilities.ICapabilityProvider;
+import net.minecraftforge.fml.relauncher.Side;
+import net.minecraftforge.fml.relauncher.SideOnly;
+import net.minecraftforge.fmp.capabilities.CapabilityWrapperRegistry;
+import net.minecraftforge.fmp.capabilities.ISlottedCapabilityProvider;
+import net.minecraftforge.fmp.event.PartEvent;
+import net.minecraftforge.fmp.multipart.ISolidPart.ISolidTopPart;
+import net.minecraftforge.fmp.network.MessageMultipartChange;
+import net.minecraftforge.fmp.network.MessageMultipartChange.Type;
+import net.minecraftforge.fmp.util.IWorldLocation;
+
+/**
+ * Helper class that contains all the logic required for an {@link IMultipartContainer} to work, as well as methods that
+ * are forwarded to each of the parts.<br/>
+ * You can implement {@link IMultipartContainer} in your {@link TileEntity} and forward the calls to an instance of this
+ * class for default multipart container logic.
```

Comment:
Do you expect EVERY SINGLE TE WHICH IMPLEMENTS THE CAP to also edit validate, invalidate and the other functions to match?

This is not an inventory. It's pretty important to have it hooked to most functions in a TE or in a Block.